File: Rock Paper Scissors/Rock Paper Scissors.py
import random

you_score = 0         # to track the user's score
computer_score = 0    # to track the computer's score

# Function to get the user's choice
def get_user_choice():
    """
    Function to get user's choice (rock, paper, or scissors)
    """
    while True:
        # remove any leading or trailing whitespaces in the input and convert it to lowercase before storing it
        user_choice = input("Enter your choice (rock, paper, or scissors): ").strip().lower()
        
        if user_choice in ['rock', 'paper', 'scissors']:
            return user_choice

        else:
            print("Invalid choice! Please enter 'rock', 'paper', or 'scissors'.")


# Function to get the computer's choice
def get_computer_choice():
    """
    Function to randomly generate computer's choice
    """
    return random.choice(['rock', 'paper', 'scissors'])


# Function to determine the winner of each round
def determine_winner(user_choice, computer_choice):
    """
    Function to determine the winner of the game
    """
    global you_score, computer_score

    if user_choice == computer_choice:
        return "It's a tie!"

    elif (user_choice == 'rock' and computer_choice == 'scissors') or \
         (user_choice == 'paper' and computer_choice == 'rock') or \
         (user_choice == 'scissors' and computer_choice == 'paper'):
         
        you_score = you_score + 1
        return "Congratulations! You win this round!"

    else:
        computer_score = computer_score + 1
        return "Computer wins this round!"


# Function to determine the winner of the match
def overall_winner(you_score, computer_score):
    if you_score > computer_score:
        return("You won the overall match")

    elif you_score < computer_score:
        return("Computer won the overall match")

    else:
        return("The match is tied")


# Main block
if __name__ == '__main__':
    print("Let's play Rock, Paper, Scissors!")
    print("This game is the best of 3!")
    print("\n")

    n = int(input("How many rounds do you want to play? : "))
    print("")
    count = 1     # to keep a track of the rounds

    while n > 0:
        print(f"Round: {count}")
        user_choice = get_user_choice()
        computer_choice = get_computer_choice()
        print(f"You chose: {user_choice}")
        print(f"Computer chose: {computer_choice}")
        print(determine_winner(user_choice, computer_choice))
        print("")
        n = n - 1
        count = count + 1
        
    print(overall_winner(you_score, computer_score))

File: Rock Paper Scissors/Rock Paper Scissors.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <string.h>
#include <ctype.h>

// Global variables for scores
int you_score = 0;
int computer_score = 0;

// Function to convert a string to lowercase
void toLower(char *str) 
{
    for (int i = 0; str[i]; i++) 
    {
        str[i] = tolower(str[i]);
    }
}

// Function to get the user's choice (rock, paper, or scissors)
int get_user_choice(char *user_choice, char *output) 
{
    toLower(user_choice);
  
    if (strcmp(user_choice, "rock") == 0 || strcmp(user_choice, "paper") == 0 || strcmp(user_choice, "scissors") == 0) 
    {
        strcpy(output, user_choice);
        return 1;
    }
  
    return 0;
}

// Function to generate the computer's choice (rock, paper, or scissors)
void get_computer_choice(char *output) 
{
    int rand_num = rand() % 3;
  
    if (rand_num == 0) 
    {
        strcpy(output, "rock");
    } 
    
    else if (rand_num == 1) 
    {
        strcpy(output, "paper");
    } 
    
    else 
    {
        strcpy(output, "scissors");
    }
}

// Function to determine the winner of the round
void determine_winner(char *user_choice, char *computer_choice, char *result) 
{
    if (strcmp(user_choice, computer_choice) == 0) 
    {
        strcpy(result, "It's a tie!");
    } 
    
    else if ((strcmp(user_choice, "rock") == 0 && strcmp(computer_choice, "scissors") == 0) ||
               (strcmp(user_choice, "paper") == 0 && strcmp(computer_choice, "rock") == 0) ||
               (strcmp(user_choice, "scissors") == 0 && strcmp(computer_choice, "paper") == 0)) 
    {
        you_score++;
        strcpy(result, "Congratulations! You win this round!");
    } 
    
    else 
    {
        computer_score++;
        strcpy(result, "Computer wins this round!");
    }
}

// Function to compare the scores and return the overall result
void compare_scores(char *result) 
{
    if (you_score < computer_score) 
    {
        strcpy(result, "Computer won the overall match!");
    } 
    
    else if (you_score > computer_score) 
    {
        strcpy(result, "You won the overall match!");
    } 
    
    else 
    {
        strcpy(result, "The match is tied!");
    }
}

// Driver Code
int main() 
{
    srand(time(0));

    printf("Let's play Rock, Paper, Scissors!\n");
    printf("How many rounds do you want to play? ");
  
    int rounds;
    scanf("%d", &rounds);

    int n = 0;
    while (n < rounds) 
    {
        printf("\nRound: %d\n", n + 1);

        while (1) 
        {
            char input[20];
            char user_choice[20];
            char computer_choice[20];
            char result[50];

            printf("Enter your choice (rock, paper, or scissors): ");
            scanf("%s", input);

            if (!get_user_choice(input, user_choice)) 
            {
                printf("Invalid choice! Please enter 'rock', 'paper', or 'scissors'.\n");
            }
            
            else 
            {
                get_computer_choice(computer_choice);
                printf("You chose: %s\n", user_choice);
                printf("Computer chose: %s\n", computer_choice);

                determine_winner(user_choice, computer_choice, result);
                printf("%s\n", result);

                n++;
                break;
            }
        }
    }

    char final_result[50];
    compare_scores(final_result);
    printf("\n%s\n", final_result);

    return 0;
}
